pkg/apis/apps/v1beta2/defaults_test.go

```diff
@@ -233,6 +233,10 @@ func TestSetDefaultStatefulSet(t *testing.T) {
 						},
 					},
 					RevisionHistoryLimit: utilpointer.Int32(10),
+					PersistentVolumeClaimRetentionPolicy: &appsv1beta2.StatefulSetPersistentVolumeClaimRetentionPolicy{
```

Comment:
Do we want to have the same test cases and feature gate toggling logic for v1beta1/v1beta2 as we do for the v1 defaults? https://github.com/kubernetes/kubernetes/blob/master/pkg/apis/apps/v1/defaults_test.go#L609

@liggitt does the defaulting logic still apply if we don't serve these versions anymore?

Reply:
defaulting applies when reading serialized data persisted in these versions from etcd

Reply:
@msau42 and I chatted offline, I've changed v1beta2/defaults_test.go to explicitly test the feature gate.

Note that v1beta1 doesn't need the same changes as it doesn't have stateful sets (at least they aren't in defaults_test).

Reply:
I asked @mattcary to create a followup PR to add some basic tests to the v1beta1 defaulting.

Reply:
In fact https://github.com/kubernetes/kubernetes/pull/116561 is ready to go after this one merges.